app/services/content_blocks_service.rb

```diff
@@ -9,8 +9,9 @@ def find_or_initialize(content_path)
   end
 
   # @return [ContentBlock]
-  def create_or_update(content_path, content_block_params)
+  def create_or_update(content_path, content_block_params, predecessor_version = nil)
     content_block = find_or_initialize content_path
+    raise OutdatedError if predecessor_version && predecessor_version != content_block.version
```

Comment:
i'd prefer a method checking if is outdated, so we could get rid of the exception. but maybe you discuss it with another person, so we find a consent?
```rb
def outdated?(version)
   version > 0 && version != content_block.version
end
```